src/material/tabs/tab-list-label.html

```diff
@@ -0,0 +1,3 @@
+<!-- Create a template for the content of the <mat-tab-list-label> so that we can grab a reference to this
+    TemplateRef and use it to render the tab label. -->
+<ng-template><ng-content></ng-content></ng-template>
```

Comment:
@crisbeto I created a commit in a different branch exploring an alternative approach we discussed. See https://github.com/byanofsky/components/commit/1cdece3dede89f60f67536e5b7b3b61b62adbd41.

The `MatTabListItem` is a component. The `MatTabListItems` are projected into the `MatTabList`, which then projects the rendered `MatTabListItem` into the `MatTabHeader`.

The `MatTabList` is injected into each `MatTabListItem` in order to access values necessary to render each `MatTabListItem`. Example values:
* [`selectedIndex`](https://github.com/byanofsky/components/commit/1cdece3dede89f60f67536e5b7b3b61b62adbd41#diff-e6227dd7ede7e8a05044787d9185f3e87040f79502e562bfe25256e8de0063efR72)
* [`disableRipple`](https://github.com/byanofsky/components/commit/1cdece3dede89f60f67536e5b7b3b61b62adbd41#diff-e6227dd7ede7e8a05044787d9185f3e87040f79502e562bfe25256e8de0063efR76)

Deriving the length of the `MatTabListItem` query list is a bit more complex ([see source](https://github.com/byanofsky/components/commit/1cdece3dede89f60f67536e5b7b3b61b62adbd41#diff-e6227dd7ede7e8a05044787d9185f3e87040f79502e562bfe25256e8de0063efR45-R63)). To prevent an "Expression changed after check" error, we query the query list length on the next tick, after the `MatTabList` content has been checked. Additionally, we must list for any change to the query list and update the set size accordingly.

The `MatTabListItem` click and focus events are proxied through to the appropriate event handler owned by `MatTabList` ([see source](https://github.com/byanofsky/components/commit/1cdece3dede89f60f67536e5b7b3b61b62adbd41#diff-e6227dd7ede7e8a05044787d9185f3e87040f79502e562bfe25256e8de0063efR100-R106)).